gc/base/standard/Scavenger.cpp

```diff
@@ -2776,8 +2776,12 @@ MM_Scavenger::shouldRememberObject(MM_EnvironmentStandard *env, omrobjectptr_t o
 	 * So we must not split, if it's indexable
 	 */
 	uintptr_t scannerFlags = GC_ObjectScanner::scanRoots | GC_ObjectScanner::indexableObjectNoSplit;
+	bool shouldRemember = false;
 
-	GC_ObjectScanner *objectScanner = getObjectScanner(env, objectPtr, &objectScannerState, scannerFlags);
+	GC_ObjectScanner *objectScanner = getObjectScanner(env, objectPtr, &objectScannerState, scannerFlags, SCAN_REASON_SHOULDREMEMBER, &shouldRemember);
+	if (shouldRemember) {
```

Comment:
now that you introduced flag shouldRemember, let's try to make the method does only one 'return shouldRemember' at the end, and whatever other changes needs to be done to eliminate other returns

Reply:
Do you mean that we change the method like `shouldRemember = scanObject(....);` and delegate will handle the whole scanning via ObjectScanner internally? Thanks

Reply:
nah, slot iteration loop should stay here. I was just hoping that number of returns could be reduced, but probably can't be much if any better than this. leave it as is